Warning with clang 16

I received an email from BDR about a warning from clang 16.

> clang 16 is being prepared for release in early March, so it is time to start reporting its issues.  See https://www.stats.ox.ac.uk/pub/bdr/clang16/ including its README.txt

From https://www.stats.ox.ac.uk/pub/bdr/clang16/httpuv.out:

```
Found the following significant warnings:
  httprequest.cpp:808:7: warning: ignoring return value of function declared with 'nodiscard' attribute [-Wunused-result]
See ‘/data/gannet/ripley/R/packages/tests-clang-trunk/httpuv.Rcheck/00install.out’ for details.
```

--- a/NEWS.md
+++ b/NEWS.md
@@ -1,5 +1,6 @@
 # httpuv 1.6.8.9000
 
+* Fixed #354: The incorrect method was called to clear a `vector`. (#355)
 
 # httpuv 1.6.8
 

--- a/src/httprequest.cpp
+++ b/src/httprequest.cpp
@@ -805,7 +805,7 @@ void HttpRequest::_parse_http_data(char* buffer, const ssize_t n) {
       if (body.size() > 0) {
         pDS->add(body);
       }
-      body.empty();
+      body.clear();
 
       pResp->writeResponse();
 

